File: FigurasGeometricas/EntidadesVb/Triangulo.vb
Public Class Triangulo
	Inherits Figura
	Implements Figura2D

	Public Property Base As Single
	Public Property Altura As Single
	Public Sub New(nombre As String, base As Single, altura As Single)
		MyBase.New(nombre)
		Me.Base = base
		Me.Altura = altura
	End Sub
	Public Overrides Function ToString() As String
		Return $"Soy {Nombre} y tengo la altura de {Altura} y la base de {Base}"
	End Function

	Public Function CalcularArea() As Single Implements Figura2D.CalcularArea
		Return (Base * Altura) / 2
	End Function
End Class

File: FigurasGeometricas/EntidadesCs/Triangulo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntidadesCs
{
	public class Triangulo : Figura, Figura2D
	{
		public float Altura { get; set; }
		public float Base { get; set; }
		public Triangulo(string nombre, float @base, float altura) : base(nombre)
		{
			Base = @base;
			Altura = altura;
		}
		public float CalcularArea()
		{
			return (Base * Altura) / 2;
		}
		public override string ToString()
		{
			return $"Soy {Nombre} y tengo la base de {Base} y la altura de {Altura}";
		}
	}
}
